### PR TITLE
feat(horaectl): initial commit

### PR DESCRIPTION
**Pull request by chunshao90**
## Rationale
Introduce command line tool for horaedb, `horaectl`.

## Detailed Changes
* Initial commit.
* Support `list clusters` and `cluster diagnose`.
```
go run main.go --meta_addr 127.0.0.1:8080 --cluster_name defaultCluster
127.0.0.1:8080(defaultCluster) > c  
+----+----------------+------------+--------------+-----------------------------+-------------------------+-------------------------+
| ID | NAME           | SHARDTOTAL | TOPOLOGYTYPE | PROCEDUREEXECUTINGBATCHSIZE | CREATEDAT               | MODIFIEDAT              |
+----+----------------+------------+--------------+-----------------------------+-------------------------+-------------------------+
|  0 | defaultCluster |          8 | static       |                  4294967295 | 2024-01-19 15:18:57.034 | 2024-01-19 15:18:57.034 |
+----+----------------+------------+--------------+-----------------------------+-------------------------+-------------------------+
127.0.0.1:8080(defaultCluster) > c d
+---------------------+-------------------------+--------------------------+-----------------------+
| UNREGISTERED_SHARDS | UNREADY_SHARDS:SHARD_ID | UNREADY_SHARDS:NODE_NAME | UNREADY_SHARDS:STATUS |
+---------------------+-------------------------+--------------------------+-----------------------+
| []                  |                         |                          |                       |
+---------------------+-------------------------+--------------------------+-----------------------+

```

## Test Plan
Manual test.

### CODE REVIEW

File: ctl/cmd/diagnose.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+package cmd
+
+import (
+	"github.com/apache/incubator-horaedb/ctl/operation"
+	"github.com/spf13/cobra"
+)
+
+// diagnoseCmd represents the diagnose command
```
Comment (ZuLiangWang) [inline ctl/cmd/diagnose.go:27]: Isn't it a bit redundant to create a file for every command? How about put interfaces of the same type together, for example `/api/v1/clusters`.

File: ctl/operation/util.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package operation
```
Comment (ShiKaiWi) [inline ctl/operation/util.go:1]: Missing license header.

File: ctl/LICENSE
Comment (ShiKaiWi) [inline ctl/LICENSE:1]: I guess there is no need to add a separate LICENSE for the sub-project? Just refer to the license under the root directory.

File: ctl/cmd/quit.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+
+	"os"
+)
+
+// quitCmd represents the quit command
```
Comment (ZuLiangWang) [inline ctl/cmd/quit.go:28]:
```suggestion
// quitCmd represents the quit command.
```
Unify the comment style, add `.`

File: ctl/LICENSE
Comment (jiacai2050) [inline ctl/LICENSE:1]: This file is not required, all sub project in this repo share the same license, which is in the root of this project.

File: ctl/cmd/diagnose.go (hunk 1 above)
Comment (chunshao90) [reply ctl/cmd/diagnose.go:27]: This file is generated by `cobra-cli`.

**Review by ShiKaiWi — APPROVED** [review]

**Pull request merged**